src/highlights/highlight_grid.ts

```diff
@@ -292,10 +292,18 @@ export class HighlightGrid {
             }
 
             const colHighlight = colHighlights[0];
-            const existingHighlight = normalHighlights.get(colHighlight.hlId);
-            if (existingHighlight !== undefined) {
+            const existingHighlights = normalHighlights.get(colHighlight.hlId) ?? [];
+            const matchingHighlight = rfind(
+                existingHighlights,
+                (hl) =>
+                    hl.endCol === col ||
+                    // double characters' start columns are offset by one (i.e. col is one higher)
+                    (HighlightGrid.isDouble(colHighlight.text) && hl.endCol === col - 1),
```

Comment:
I don't get it. Wasn't the highlight already processed?

Reply:
This is actually how I fixed this test

https://github.com/vscode-neovim/vscode-neovim/pull/1915/files#diff-170234a32b0ecf69f0c4d9fa8465ac6542ecc30db6ada74a9dd1754e16a3b96dL200-L208

In that test, once we get to the ` `, `col` is 13. Once we hit the rocket emoji, `col` jumps to 15, due to the fact the char is double-wide, and `hl.endCol` at this point is 14, because we just processed column 13. If we simply checked `hl.endCol === col`, this would produce two separate highlight ranges, so we make the adjustment for this double-wide char.

Perhaps there's some argument to be made that the previously calculated highlight is wrong? Stepping through `processHighlightCellsEvent`, the additional width is coming from this check, which I suppose is right by coincidence, as ` "🚀".length === 2`

https://github.com/vscode-neovim/vscode-neovim/blob/3c82cc01e69e0817eb92f7093844136a3038bcff/src/highlights/highlight_grid.ts#L151-L152

Reply:
On second thought, I think you're right... I think this is just compensating for a bug in calculating the highlight ranges. I'll go ahead and revert this. I'm actually tracking down another bug (predating this PR) that might stem from the same problem, so I'll fix this there if I can.